### PR TITLE
chore: migrate Android unit tests to GH actions

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
This PR migrates Android unit tests to GH actions. It largely copies the actions from next with some manual modifications (like only running against Flutter stable, not beta). I looked at the CircleCI config and made sure to test the same packages as that aspect is a little different from next.

Removing CircleCI equivalents should be separate PR. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_core/android/src/test/kotlin/com/amazonaws/amplify/amplify_core/AtomicResultTest.kt [hunk 1]
```diff
@@ -66,21 +63,4 @@ class AtomicResultTest {
         atomicResult.success(null)
         verify(mockResult, times(1)).success(any())
     }
-
-    @Test
-    fun multipleConcurrentRepliesAreNotSent() = coroutinesTestRule.testDispatcher.runBlockingTest {
```
Comment (ragingsquirrel3) [inline packages/amplify_core/android/src/test/kotlin/com/amazonaws/amplify/amplify_core/AtomicResultTest.kt:71]: this test was flaky on gh actions, not sure why (same issue on next). I just removed it here (and on next) but open to keeping if anyone has suggestions. I didn't think it was worth investigation.

**Review by Jordan-Nelson — DISMISSED** [review]

File: .github/workflows/amplify_analytics_pinpoint.android.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: amplify_analytics_pinpoint Android
+on:
+  push:
+    branches:
+      - main
+      - stable
+      - next
```
Comment (NikaHsn) [inline .github/workflows/amplify_analytics_pinpoint.android.yaml:7]: do we want this for next branch?

Comment (NikaHsn) [inline .github/workflows/amplify_analytics_pinpoint.android.yaml:7]: do we want it for next branch?

Comment (ragingsquirrel3) [reply .github/workflows/amplify_analytics_pinpoint.android.yaml:7]: nope! Good catch, will remove.

**Review by NikaHsn — APPROVED** [review]

**Review by Jordan-Nelson — APPROVED** [review]

**Pull request merged**